back/app/controllers/web_api/v1/notifications_controller.rb

```diff
@@ -17,12 +17,10 @@ def index
     end
 
     @notifications = paginate @notifications
-    render json: linked_json(
-      @notifications,
-      WebApi::V1::Notifications::NotificationSerializer,
-      params: fastjson_params,
-      serializers: NotificationService.new.serializers
-    )
+    render json: {
+      data: serialize_heterogeneous_collection(@notifications, serializers_classes),
+      links: page_links(@notifications)
+    }
```

Comment:
Changes in this controller to deal with the heterogenous collections, and serializing single records which could be any of multiple types.

This is the only controller where we seemed to do anything similar previously.